src/game/process/horde/Brain.js

```diff
@@ -34,22 +34,51 @@ export class Brain {
     let targetPlayer = null;
     // console.log(` ==== ${monster.walletAddress} =====`);
 
-    if (this._isAlreadyAttackingPlayer()) {
-      if (this._isPlayerWithIdAlive(monster.attackedPlayerId)) {
-        targetPlayer = this._state.players[monster.attackedPlayerId];
-      } else {
+    /**
+     * if monster is both already attacking some player and at the same time is being
+     * attacked by another player AND the monster is aggressive enough
+     * - it will switch to the player that is attacking him
+     * UNLESS - the monster is attacking a digging player - in such
+     * case he is focusing on him
+     */
+    if (this._isAlreadyAttackingPlayer() && !this._isAttackingDigger() && this._isBeingAttackedByOtherPlayer()) {
+      const chanceToFightBack = Math.random(++this._state.randomCounter);
+      if (
+        chanceToFightBack <= monster.stats.aggressiveness &&
+        this._isPlayerWithIdAlive(monster.attackingPlayerId) &&
+        this._isPlayerInAttackRange(monster.attackingPlayerId)
+      ) {
+        targetPlayer = this._state.players[monster.attackingPlayerId];
         monster.attackedPlayerId = null;
       }
-    } else if (this._isBeingAttackedByOtherPlayer()) {
-      if (this._isPlayerWithIdAlive(monster.attackingPlayerId)) {
-        targetPlayer = this._state.players[monster.attackingPlayerId];
-      } else {
-        monster.attackingPlayerId = null;
+    }
+
+    if (!targetPlayer) {
+      if (this._isAlreadyAttackingPlayer()) {
+        if (this._isPlayerWithIdAlive(monster.attackedPlayerId)) {
+          targetPlayer = this._state.players[monster.attackedPlayerId];
+          if (!this._isAttackingDigger() && this._anyActiveDiggers()) {
+            const chanceToSwitchToDigger = Math.random(++this._state.randomCounter);
+            if (chanceToSwitchToDigger <= monster.stats.aggressiveness) {
+              targetPlayer = null;
```

Comment:
we're just resetting targetPlayer here - switiching to digger is guaranteed by the call to `_findPlayerToAttack` in line 79.